zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -273,15 +274,15 @@ public static Either<ProblemDetail, DocumentCreateRequest> toDocumentCreateReque
   }
 
   public static Either<ProblemDetail, CreateUserRequest> toCreateUserRequest(
-      final UserRequest request) {
+      final UserRequest request, final PasswordEncoder passwordEncoder) {
     return getResult(
         validateUserCreateRequest(request),
         () ->
             new CreateUserRequest(
                 request.getUsername(),
                 request.getName(),
                 request.getEmail(),
-                request.getPassword()));
+                passwordEncoder.encode(request.getPassword())));
```

Comment:
💭 I remember a discussion about it. I thought we said that Zeebe should not be in charge of encoding the password, it should just store and export the password as it is. Identyty should be in charge of provide an encrypted password

Reply:
But if you encode the password here, you encode it for every request. The default user is not created via REST API

Reply:
Oh sorry, I thought this comment was in the `DefaultUserCreator` 🤦 Forgot I had to add this here as well. I had to add this back since I accidentally removed it in a refactoring.

I'll clarify further with Identity, but will merge this in the meantime. I think it makes sense to do it here as this is a public endpoint. Users can call this without going through Identity. As a result, we cannot guarantee they will use the same encoding we need. If they'd use something different from us we cannot match the password upon login.

Reply:
Ok got it